packages/bundler/src/modules/BundleManagerRIP7560.ts

```diff
@@ -0,0 +1,98 @@
+import Debug from 'debug'
+import { BigNumber, BigNumberish } from 'ethers'
+import { JsonRpcProvider } from '@ethersproject/providers'
+
+import {
+  getRIP7560TransactionHash,
+  IEntryPoint, OperationBase,
+  OperationRIP7560,
+  StorageMap
+} from '@account-abstraction/utils'
+import { IValidationManager } from '@account-abstraction/validation-manager'
+
+import { IBundleManager } from './IBundleManager'
+import { MempoolManager } from './MempoolManager'
+import { ReputationManager } from './ReputationManager'
+import { BundleManager, SendBundleReturn } from './BundleManager'
+import { EventsManager } from './EventsManager'
+
+const debug = Debug('aa.exec.cron')
+
+export class BundleManagerRIP7560 extends BundleManager implements IBundleManager {
+  sentBundles: SendBundleReturn[] = []
+  lastScannedBlock: number = 0
+
+  constructor (
+    _entryPoint: IEntryPoint | undefined,
+    readonly eventsManager: EventsManager,
+    readonly mempoolManager: MempoolManager,
+    readonly validationManager: IValidationManager,
+    readonly reputationManager: ReputationManager,
+    readonly beneficiary: string,
+    readonly minSignerBalance: BigNumberish,
+    readonly maxBundleGas: number,
+    readonly conditionalRpc: boolean,
+    readonly mergeToAccountRootHash: boolean = false,
+    readonly provider: JsonRpcProvider
+  ) {
+    super(
+      _entryPoint, eventsManager, mempoolManager, validationManager,
+      reputationManager, beneficiary, minSignerBalance, maxBundleGas,
+      conditionalRpc, mergeToAccountRootHash
+    )
+    this.provider = provider
+  }
+
+  async sendNextBundle (): Promise<SendBundleReturn | undefined> {
+    await this.handlePastEvents()
+
+    const [bundle] = await this.createBundle()
+    if (bundle.length === 0) {
+      debug('sendNextBundle - no bundle to send')
+    } else {
+      return await this.sendBundle(bundle, '', {})
+    }
+  }
+
+  async handlePastEvents (): Promise<any> {
+    const bundlesToClear: string[] = []
+    for (const bundle of this.sentBundles) {
+      // TODO: apply reputation changes based on the Bundle Stats API response
+      const bundleStats = await this.provider.send('eth_getRip7560BundleStatus', [bundle.transactionHash])
+      if (bundleStats != null) {
+        bundlesToClear.push(bundle.transactionHash)
+      }
+
+      for (const operationHash of bundle.userOpHashes) {
+        this.mempoolManager.removeUserOp(operationHash)
+      }
+    }
+    for (const bundleId of bundlesToClear) {
+      this.sentBundles = this.sentBundles.filter(it => it.transactionHash !== bundleId)
+    }
+  }
+
+  async sendBundle (userOps: OperationBase[], _beneficiary: string, _storageMap: StorageMap): Promise<any> {
```

Comment:
deprecated. should have "pullBundle" instead.

Reply:
This is extracted into a separate issue:
https://opengsn.atlassian.net/browse/AA-381